pkg/scheduler/framework/plugins/volumezone/volume_zone_test.go

```diff
@@ -423,17 +451,134 @@ func TestWithBinding(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			state := framework.NewCycleState()
 			node := &framework.NodeInfo{}
 			node.SetNode(test.Node)
 			p := &VolumeZone{
 				pvLister,
 				pvcLister,
 				scLister,
 			}
-			gotStatus := p.Filter(context.Background(), nil, test.Pod, node)
-			if !reflect.DeepEqual(gotStatus, test.wantStatus) {
-				t.Errorf("status does not match: %v, want: %v", gotStatus, test.wantStatus)
+			_, gotPreFilterStatus := p.PreFilter(ctx, state, test.Pod)
+			if !gotPreFilterStatus.IsSuccess() {
+				if diff := cmp.Diff(gotPreFilterStatus, test.wantStatus); diff != "" {
+					t.Errorf("status does not match (-want,+got):\n%s", diff)
+				}
+			} else {
+				gotStatus := p.Filter(ctx, state, test.Pod, node)
+				if diff := cmp.Diff(gotStatus, test.wantStatus); diff != "" {
+					t.Errorf("status does not match (-want,+got):\n%s", diff)
+				}
 			}
 		})
 	}
 }
+
+func BenchmarkVolumeZone(b *testing.B) {
+	tests := []struct {
+		Name      string
+		Pod       *v1.Pod
+		NumPV     int
+		NumPVC    int
+		NumNodes  int
+		PreFilter bool
+	}{
+		{
+			Name:      "with prefilter",
+			Pod:       createPodWithVolume("pod_0", "Vol_Stable_0", "PVC_Stable_0"),
+			NumPV:     10000,
+			NumPVC:    10000,
```

Comment:
given that the fake lister does a linear search, this might not be appropriate. Maybe 2 or 3 is enough?

Reply:
ah, sounds good

Reply:
Maybe do a quick experiment with less PVs to make sure the runtime is not too different.

Reply:
I ran with both number set to 10 on my  computer before, seems  there is not much difference.

with 1000:
```
BenchmarkVolumeZone/with_prefilter
BenchmarkVolumeZone/with_prefilter-16         	   19843	     64622 ns/op
BenchmarkVolumeZone/without_prefilter
BenchmarkVolumeZone/without_prefilter-16      	    2204	    536765 ns/op
```

with 10:
```
BenchmarkVolumeZone/with_prefilter
BenchmarkVolumeZone/with_prefilter-16         	   21940	     57998 ns/op
BenchmarkVolumeZone/without_prefilter
BenchmarkVolumeZone/without_prefilter-16      	    1904	    555315 ns/op
```